app/models/bookmark.rb

```diff
@@ -8,7 +8,11 @@ class Bookmark < ActiveRecord::Base
   def  httpset
     if self.url =~ /\Ahttp:\/\/|\Ahttps:\/\//i
     else
-      self.url = "http://"+ self.url
+      if self.url.present?
```

Comment:
What do you think about validating the presence of URL's to save the bookmark? They are pretty much worthless without a URL. You could do it like this:

`validates :url, presence: true`

Then you wouldn't need the `if self.url.present?` check.